main/src/library/Process.flix

```diff
@@ -14,22 +15,73 @@
  *  limitations under the License.
  */
 
+///
+/// Handles the created Process
+///
+enum ProcessHandle({stop = Unit -> Unit \ {Exec, IO}}, {exitValue = Unit -> Int32 \ {Exec, IO}}, 
+    {isAlive = Unit -> Bool \ {Exec, IO}})
+
+mod ProcessHandle {
+
+    ///
+    /// Stops the process
+    ///
+    pub def stop(ph: ProcessHandle) : Unit \ {IO, Exec} = match ph {
+        case ProcessHandle.ProcessHandle({stop}, _, _) => stop()
+    }
+
+    ///
+    /// Returns the exit value of the process
+    ///
+    pub def exitValue(ph: ProcessHandle) : Int32 \ {IO, Exec} = match ph {
+        case ProcessHandle.ProcessHandle(_, {exitValue}, _) => exitValue()
+    }
+
+    ///
+    ///  Returns status of the process, if it alive or not
+    ///
+    pub def isAlive(ph: ProcessHandle) : Bool \ {IO, Exec} = match ph {
+        case ProcessHandle.ProcessHandle(_, _, {isAlive}) => isAlive()
+    }
+}
+
 ///
 /// An effect used to start a process outside the JVM.
 ///
 eff Process {
 
     ///
-    /// Immediately executes the command `cmd` passing the arguments `args`.
+    /// Immediately executes the command `cmd` with the arguments `args`, by the path `cwd` and with the environmental `env`
     ///
-    def exec(cmd: String, args: List[String]): Unit
-
+    def execWithCwdAndEnv(cmd: String, args: List[String], cwd: String, env: Map[String, String]): ProcessHandle
 }
 
+
+
 mod Process {
 
     import java.lang.ProcessBuilder
     import java.lang.Runtime
+    import java.io.{File => JFile}
+
+
+    ///
+    /// Executes the command `cmd` with the arguments `args`
+    ///
+    pub def exec(cmd: String, args: List[String]): ProcessHandle \ Process =
+        Process.execWithCwdAndEnv(cmd, args, "", Map.empty())
```

Comment:
I think there should be a difference between setting cwd to the empty string and the actual cwd?

Reply:
I think it is better to leave it like that because the user can give an empty string as an argument anyway.

If we decide to separate them, it will probably be better to move `execWithEnv` to the `Process` effect itself since `execWithEnv` and `execWithCwdAndEnv` have kinda different behavior. But for me, it seems unnecessary because the user can provide an empty string as a `cwd`, and we will need to address it in the second function, which will make the work of the first one.

Reply:
Why is it correct to set CWD to the empty string?

Reply:
I c. In that case, it makes sense to separate them and throw an error in the second function. And I think it still makes sense to move `execWithEnv ` to the effect since it will have a different behavior.

Or we can change the type of `cwd` to `Option[String]` and it can solve that issue

Reply:
I think it would make sense to make cwd Option[String]